### PR TITLE
i2pd: new version

### PR DESCRIPTION
**Pull request by hfsfox**

### CODE REVIEW

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libboost_filesystem$secondaryArchSuffix >= $boostMinumumVersion
+	devel:libboost_system$secondaryArchSuffix >= $boostMinumumVersion
+	devel:libboost_date_time$secondaryArchSuffix >= $boostMinumumVersion
+	devel:libboost_program_options$secondaryArchSuffix >= $boostMinumumVersion
+	devel:libboost_config$secondaryArchSuffix >= $boostMinumumVersion	
+	devel:libcrypto$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:52]: Trailing whitespace at the end.

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	i2pd$secondaryArchSuffix
+	cmd:i2pd$secondaryArchSuffix
+	lib:libi2pd$secondaryArchSuffix
+	lib:libi2pdclient$secondaryArchSuffix
+	lib:libi2pdlang$secondaryArchSuffix
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libboost_filesystem$secondaryArchSuffix >= $boostMinumumVersion
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:36]: Sort alphabetically here and in other sections.

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	cmd:lrelease$secondaryArchSuffix >= 5
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:linguist$secondaryArchSuffix >= 5
+	cmd:make
+	"	
+
+BUILD()
+{
+	export LDFLAGS="-lnetwork -lbsd" 
+	cmake -Sbuild -Bbuild $cmakeDirArgs -DCMAKE_BUILD_TYPE=Release -DBoost_USE_STATIC_LIBS=OFF -DBoost_USE_STATIC_RUNTIME=OFF -DBUILD_SHARED_LIBS=ON
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:67]: This line is too long.

**Review by diversys — CHANGES_REQUESTED** [review]

**Comment by korli** [comment]
Please remove the old version recipe and patchset

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+(websites, instant messengers, chat-servers).\
+I2P allows people from all around the world to communicate and share information without restrictions.
+"
+HOMEPAGE="https://i2pd.website/"
+COPYRIGHT="i2pd developers"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/PurpleI2P/i2pd/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="d8e07f78cc9f1ba65e8460db27c649dd0cfdd3ba334725f8d6f9ee815cb40e68"
+PATCHES="$portName-$portVersion.patchset"
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:18]: As mentioned in the previous PR, don't use `$portName` here. This failed last time on buildmaster 32bit also.

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	"
+
+BUILD()
+{
+	export LDFLAGS="-lnetwork -lbsd"
+	cmake -Sbuild -Bbuild $cmakeDirArgs \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_SHARED_LIBS=ON \
+	-DBoost_USE_STATIC_LIBS=OFF \
+	-DBoost_USE_STATIC_RUNTIME=OFF
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:71]: Use indents here:
```
	cmake -Sbuild -Bbuild $cmakeDirArgs \
		-DCMAKE_BUILD_TYPE=Release \
		-DBUILD_SHARED_LIBS=ON \
		-DBoost_USE_STATIC_LIBS=OFF \
		-DBoost_USE_STATIC_RUNTIME=OFF
```

Comment (Begasus) [reply net-vpn/i2pd/i2pd-2.49.0.recipe:71]: Missing "make -C build $jobARgs"

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+}
+INSTALL()
+{
+	cd build
+	make install
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:76]: One line should do this:
```diff
-	cd build
-	make install
+	make -C build install
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe (hunk 5 above)
Comment (korli) [reply net-vpn/i2pd/i2pd-2.49.0.recipe:71]: Shouldn't be `-S.` ?

File: net-vpn/i2pd/patches/i2pd-2.49.0.patchset [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
++
+ #ifdef _WIN32
+ #include <stdlib.h>
+ #include <string.h>
+@@ -190,8 +200,10 @@ namespace util
+ 		}
+ 		#pragma warning(pop)
+ 	#else
++	#if !defined(__HAIKU__)
+ 		pthread_setname_np(pthread_self(), name);
```
Comment (korli) [inline net-vpn/i2pd/patches/i2pd-2.49.0.patchset:80]: pthread_setname_np should be available on Haiku (see https://github.com/haiku/haiku/commit/44cceee67e056d8e34cb017d5f5c07b9fac874c0)

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 8]
```diff
@@ -0,0 +1,77 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.\
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.\
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.\
+I2P client is a software used for building and using anonymous I2P networks. \ 
+Such networks are commonly used for anonymous peer-to-peer applications \
+(filesharing, cryptocurrencies) and anonymous client-server applications \ 
+(websites, instant messengers, chat-servers).\
+I2P allows people from all around the world to communicate and share information without restrictions.
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:10]:
```suggestion
I2P allows people from all around the world to communicate and share information without \
restrictions".
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 9]
```diff
@@ -0,0 +1,77 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.\
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.\
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.\
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:5]:
```suggestion
don't reveal their real IP addresses.
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 10]
```diff
@@ -0,0 +1,77 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.\
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.\
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:3]:
```suggestion
I2P (Invisible Internet Protocol) is a universal anonymous network layer.
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.\
+I2P client is a software used for building and using anonymous I2P networks. \ 
+Such networks are commonly used for anonymous peer-to-peer applications \
+(filesharing, cryptocurrencies) and anonymous client-server applications \ 
+(websites, instant messengers, chat-servers).\
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:9]:
```suggestion
(websites, instant messengers, chat-servers).
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 12]
```diff
@@ -0,0 +1,78 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.\
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:2]:
```suggestion
DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 13]
```diff
@@ -0,0 +1,78 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.
+I2P client is a software used for building and using anonymous I2P networks. \
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:6]:
```suggestion
I2P client is a software used for building and using anonymous I2P networks. \
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 14]
```diff
@@ -0,0 +1,78 @@
+SUMMARY="I2P: End-to-End encrypted and anonymous Internet"
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.
+I2P client is a software used for building and using anonymous I2P networks. \ 
+Such networks are commonly used for anonymous peer-to-peer applications \
+(filesharing, cryptocurrencies) and anonymous client-server applications \
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:8]:
```suggestion
(filesharing, cryptocurrencies) and anonymous client-server applications \
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+DESCRIPTION="i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.
+I2P client is a software used for building and using anonymous I2P networks. \ 
+Such networks are commonly used for anonymous peer-to-peer applications \
+(filesharing, cryptocurrencies) and anonymous client-server applications \ 
+(websites, instant messengers, chat-servers).
+I2P allows people from all around the world to communicate and share information without \
+restrictions".
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:11]:
```suggestion
restrictions."
```

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+I2P (Invisible Internet Protocol) is a universal anonymous network layer.
+All communications over I2P are anonymous and end-to-end encrypted, participants \
+don't reveal their real IP addresses.
+I2P client is a software used for building and using anonymous I2P networks. \
+Such networks are commonly used for anonymous peer-to-peer applications \
+(filesharing, cryptocurrencies) and anonymous client-server applications \
+(websites, instant messengers, chat-servers).
+I2P allows people from all around the world to communicate and share information without \
+restrictions."
+"
```
Comment (diversys) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:12]:
```suggestion

```

File: net-vpn/i2pd/i2pd-2.49.0.recipe (hunk 5 above)
Comment (Begasus) [reply net-vpn/i2pd/i2pd-2.49.0.recipe]: CMakeLists.txt and needed files are provided in "build" directory. :)

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+{
+	export LDFLAGS="-lnetwork -lbsd"
+	cmake -Sbuild -Bbuild $cmakeDirArgs \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DBUILD_SHARED_LIBS=ON \
+		-DBoost_USE_STATIC_LIBS=OFF \
+		-DBoost_USE_STATIC_RUNTIME=OFF
+	make -C build $jobArgs
+}
+INSTALL()
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:79]: Empty line before INSTALL()

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+
+PROVIDES="
+	i2pd$secondaryArchSuffix
+	cmd:i2pd$secondaryArchSuffix
+	lib:libi2pd$secondaryArchSuffix
+	lib:libi2pdclient$secondaryArchSuffix
+	lib:libi2pdlang$secondaryArchSuffix
+	"
+
+REQUIRES="
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:34]: No empty line between PROVIDES and BUILD_REQUIRES

File: net-vpn/i2pd/i2pd-2.49.0.recipe [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+	haiku$secondaryArchSuffix
+	lib:libboost_date_time$secondaryArchSuffix >= $boostMinumumVersion
+	lib:libboost_filesystem$secondaryArchSuffix >= $boostMinumumVersion
+	lib:libboost_program_options$secondaryArchSuffix >= $boostMinumumVersion
+	lib:libboost_system$secondaryArchSuffix >= $boostMinumumVersion
+	lib:libcrypto$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
```
Comment (Begasus) [inline net-vpn/i2pd/i2pd-2.49.0.recipe:44]: Empty line before BUILD_REQUIRES

**Review by Begasus — CHANGES_REQUESTED** [review]

File: net-vpn/i2pd/i2pd-2.49.0.recipe (hunk 19 above)
Comment (Begasus) [reply net-vpn/i2pd/i2pd-2.49.0.recipe:44]: See https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines

File: net-vpn/i2pd/i2pd-2.49.0.recipe (hunk 5 above)
Comment (jmairboeck) [reply net-vpn/i2pd/i2pd-2.49.0.recipe]: Then probably another directory name should be used as build directory. With cmake, a separate build directory is usually preferred.

**Pull request closed without merging**